book/src/spec.md

```diff
@@ -0,0 +1,628 @@
+> be useful, as physical streams also carry metadata.
+
+The letter \\(E\\) is used as a shorthand for the total number of bits in the
+element; that is, the sum of the field bit count over all fields in the
+element.
+
+#### Field name
+
+The name of each field is a string consisting of letters, numbers, and/or
+underscores.
+
+The name cannot start or end with an underscore.
+
+The name cannot start with a digit.
+
+> It is illegal to start or end with an underscore or start with a number to
+> prevent confusion when the name is prefixed to form the signal name, and
+> for compatibility with VHDL.
+
+The name cannot be empty.
+
+> Fields may be auto-named by the next layer of the spec based on the abstract
+> type if necessary.
+
+The name must be case-insensitively unique within the set of named fields.
+
+> The identifier is case-insensitive because compatibility with VHDL is
+> desired.
+
+#### Bitcount
+
+The bit count cannot be zero.
+
+> A lot of synthesis tools out there do not handle null ranges very well.
+
+### Number of element lanes (N)
+
+The data described by the element content is physically replicated this many
+times, allowing multiple elements to be transferred per stream handshake.
+\\(N\\) must be an integer greater than or equal to one.
+
+### Dimensionality (D)
+
+This parameter specifies how many `last` bits are needed to represent the data.
+It must be an integer greater than or equal to zero.
+
+> Intuitively, each list nesting level adds one to this number. For instance,
+> to stream two-dimensional lists, two last bits will be needed: one to mark
+> the boundaries of the inner list, and one to mark the boundary of each
+> two-dimensional list.
+
+### Complexity (C)
+
+This parameter specifies what guarantees the source makes about how elements
+are transferred. Equivalently, it specifies how many assumptions the sink can
+make.
+
+| Higher C                        | Lower C                        |
+|---------------------------------|--------------------------------|
+| Fewer guarantees made by source | More guarantees made by source |
+| Source is easier to implement   | Source is harder to implement  |
+| Sink can make fewer assumptions | Sink can make more assumptions |
+| Sink is harder to implement     | Sink is easier to implement    |
+
+The complexity levels and signals are defined such that a source with
+complexity \\(C_a\\) can be connected to a sink with complexity
+\\(C_b \ge C_a\\).
+
+The complexity number is defined as a nonempty, period-separated list of
+nonnegative integers. A complexity number is higher than another when the
+leftmost integer is greater, and lower when the leftmost integer is lower. If
+the leftmost integer is equal, the next integer is checked recursively. If one
+complexity number has more entries than another, the shorter number is padded
+with zeros on the right.
+
+> That is, it works a bit like a version number. The idea behind this is to
```

Comment:
It's there to signal that the two paragraphs are connected, otherwise what "it" refers to is more vague. IMO it's much less pleasing to read without it (and the quote blocks are intended to be less formal anyway).